### PR TITLE
Azure PKE tags refactor

### PR DESCRIPTION
**Pull request by siliconbrain**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| API breaks?     | no
| Deprecations?   | no
| Related tickets | -
| License         | Apache 2.0

### What's in this PR?
Use `Tag` struct instead of string pairs.

### Why?
Make working with tags easier.

### Checklist
- [x] Implementation tested (with at least one cloud provider)
- [x] Error handling code meets the [guideline](https://github.com/banzaicloud/pipeline/blob/master/docs/error-handling-guide.md)
- [x] Logging code meets the guideline (TODO)

### CODE REVIEW

File: internal/providers/azure/pke/workflow/azure.go [hunk 1]
```diff
@@ -50,43 +50,54 @@ func (f *AzureClientFactory) New(organizationID uint, secretID string) (*pkgAzur
...
 
 func getOwnershipTagKey(clusterName string) string {
 	return fmt.Sprintf("kubernetesCluster-%s", clusterName)
 }
 
-func tagsFrom(key, value string) map[string]string {
-	return map[string]string{
-		key: value,
+func tagsToMap(tags ...Tag) map[string]string {
+	m := make(map[string]string)
```
Comment (bonifaido) [inline internal/providers/azure/pke/workflow/azure.go]: Could be `m := make(map[string]string, len(tags))`

File: internal/providers/azure/pke/workflow/azure.go [hunk 2]
```diff
@@ -50,43 +50,54 @@ func (f *AzureClientFactory) New(organizationID uint, secretID string) (*pkgAzur
...
+	return hasTag(tags, getOwnedTag(clusterName))
 }
 
 func HasSharedTag(clusterName string, tags map[string]string) bool {
-	key, val := getSharedTag(clusterName)
-	return hasTag(tags, key, val)
+	return hasTag(tags, getSharedTag(clusterName))
 }
 
 func RemoveSharedTag(tags map[string]string, clusterName string) map[string]string {
```
Comment (orymate) [inline internal/providers/azure/pke/workflow/azure.go]: shouldn't this either work on a copy of the input map, or have no return value? maps are not like slices in this regard

File: internal/providers/azure/pke/workflow/azure.go [hunk 3]
```diff
@@ -50,43 +50,54 @@ func (f *AzureClientFactory) New(organizationID uint, secretID string) (*pkgAzur
...
 }
 
 func getOwnershipTagKey(clusterName string) string {
 	return fmt.Sprintf("kubernetesCluster-%s", clusterName)
 }
 
-func tagsFrom(key, value string) map[string]string {
-	return map[string]string{
-		key: value,
+func tagsToMap(tags ...Tag) map[string]string {
```
Comment (orymate) [inline internal/providers/azure/pke/workflow/azure.go]:
```
func (t Tag) Map() map[string]string {
```
would be more legible for all current uses

File: internal/providers/azure/pke/workflow/azure.go (hunk 2 above)
Comment (siliconbrain) [reply internal/providers/azure/pke/workflow/azure.go]: returning the modified map makes the function "chainable", e.g.: `tags = RemoveSomeOtherTag(RemoveSharedTag(tags, "my-cluster"), "my-other-key")`

**Pull request merged**